src/Microsoft.ML.Onnx/ProtabilityCatalog.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Model.Onnx;
+using Microsoft.ML.UniversalModelFormat.Onnx;
+
+namespace Microsoft.ML
+{
+    public static class ProtabilityCatalog
+    {
+        /// <summary>
+        /// Convert the specified <see cref="ITransformer"/> to ONNX format. Note that ONNX uses Google's Protobuf so the returned value is a Protobuf object.
+        /// </summary>
+        /// <param name="catalog">A field in <see cref="MLContext"/> which this function associated with.</param>
+        /// <param name="transform">The <see cref="ITransformer"/> that will be converted into ONNX format.</param>
+        /// <param name="inputData">The input of the specified transform.</param>
+        /// <returns></returns>
+        public static ModelProto ConvertToOnnx(this ModelOperationsCatalog.PortabilityTransforms catalog, ITransformer transform, IDataView inputData)
+        {
+            var env = new MLContext(seed: 1);
+            var ctx = new OnnxContextImpl(env, "model", "ML.NET", "0", 0, "com.microsoft", OnnxVersion.Stable);
+            var outputData = transform.Transform(inputData);
+            IDataView root = null;
+            IDataView sink = null;
+            LinkedList<ITransformCanSaveOnnx> transforms = null;
+            using (var ch = (env as IChannelProvider).Start("ONNX conversion"))
+                SaveOnnxCommand.GetPipe(ctx, ch, outputData, out root, out sink, out transforms);
+
+            return SaveOnnxCommand.ConvertTransformListToOnnxModel(ctx, root, sink, transforms, null, null);
```

Comment:
>return SaveOnnxCommand.ConvertTransformListToOnnxModel(ctx, root, sink, transforms, null, null); [](start = 12, length = 96)

Would it be inappropriate to make this part of the `using` block? It just seems a little odd to consider that `ConvertTransformListToOnnxModel` is *separate* from a channel quite explicitly identified as being "ONNX Conversion", since this call, right here, is precisely where that conversion is happening? So why is it not in the using block of that channel? Just seems a bit off.

Once you do that, you can also get rid of your `root` and `sink`, and change `out sink, out transforms` to `out var sink, out var transforms` to clean up your code a bit...